test/connectors/TaxEnrolmentsConnectorSpec.scala

```diff
@@ -79,108 +90,110 @@ class TaxEnrolmentsConnectorSpec extends AnyWordSpec with Matchers with WireMock
     )
   ))
 
+  val expectedErrorMessage = "No content to map due to end-of-input"
 
-  private def wiremock(payload: String, expectedStatus: Int, url: String) =
+  private def wiremock(payload: String, expectedStatus: Int, url: String, body: String = "") = {
     server.stubFor(
       put(urlEqualTo(url))
         .withHeader(CONTENT_TYPE, containing("application/json"))
         .withRequestBody(equalTo(payload))
         .willReturn(
           aResponse()
             .withStatus(expectedStatus)
+            .withBody(body)
         )
     )
+  }
 
-  "TaxEnrolmentsConnector" when {
-
-    "taxable" must {
-
-      "returns 204 NO_CONTENT" in {
-
-        wiremock(
-          payload = taxableRequest,
-          expectedStatus = NO_CONTENT,
-          url = taxableEnrolmentUrl
-        )
-
-        connector.enrol(TaxEnrolmentsRequest(utr)) map { response =>
-          response mustBe EnrolmentCreated
-        }
-
-      }
-
-      "returns 400 BAD_REQUEST" in {
-
-        wiremock(
-          payload = taxableRequest,
-          expectedStatus = BAD_REQUEST,
-          url = taxableEnrolmentUrl
-        )
-
-        connector.enrol(TaxEnrolmentsRequest(utr)).value map { response =>
-          response mustBe Left(errors.UpstreamTaxEnrolmentsError)
-        }
-
-      }
-      "returns 401 UNAUTHORIZED" in {
-
-        wiremock(
-          payload = taxableRequest,
-          expectedStatus = UNAUTHORIZED,
-          url = taxableEnrolmentUrl
-        )
-
-        connector.enrol(TaxEnrolmentsRequest(utr)).value map { response =>
-          response mustBe Left(errors.UpstreamTaxEnrolmentsError)
-        }
+  private def awaitResult(request: TaxEnrolmentsRequest): Either[TrustErrors, EnrolmentResponse] =
```

Comment:
Instead of using `awaitResult` it is simpler to use the following in the test, L149 can be:
```
            whenReady(connector.enrol(TaxEnrolmentsRequest(identifier)).value, Timeout(Span.Max)){
              result => result mustBe UpstreamTaxEnrolmentsError(s"HTTP response ${responseBody}")
            }
```
the test must extend the `ScalaFutures` trait to gain access to the `whenReady` func.

Reply:
`whenReady` not always works as you are expecting especially in our tests where we mutate the same global object/database/in memory data, its better to use raw value even between `Await.result` and `.futureValue` there is a big difference how they are implemented , most simple option and easier to understand is `Await.result`

hope this make sense if not I can explain in more details with some examples

Reply:
Ok lets use awaitResult instead 👍 (ignore this  comment)